src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -536,6 +539,39 @@ namespace Abs {
 
 } // namespace Abs
 
+#define create_exp_macro(X)                                                               \
+namespace X {                                                                             \
+    static inline ASR::expr_t* eval_##X(Allocator &al, const Location &loc,               \
+            Vec<ASR::expr_t*> &args) {                                                    \
+        LCOMPILERS_ASSERT(ASRUtils::all_args_evaluated(args));                            \
+        ASR::ttype_t* t = ASRUtils::expr_type(args[0]);                                   \
+        if (ASRUtils::is_real(*t)) {                                                      \
+            double val = ASR::down_cast<ASR::RealConstant_t>(args[0])->m_r;               \
+            return EXPR(ASR::make_RealConstant_t(al, loc, val, t));                       \
```

Comment:
Here you are just extracting the value and returning the same thing. Instead, we have to compute the compile time value. 
For example, if you have extracted the value, compute `exp()` or `exp2()` or `expm1` of that value and then return it.
See: https://github.com/lcompilers/lpython/blob/268cec63c8c0a4efc8dc05876ae727106a4623a3/src/libasr/pass/intrinsic_function_registry.h#L306-L308

Reply:
Thanks for pointing it out, fixed it in the latest commit.